src/pixel.rs

```diff
@@ -0,0 +1,81 @@
+use std::ops::{Index, IndexMut};
+use std::slice;
+
+type Point = (usize, usize);
+
+/// A wrapper struct for a a one-dimensional vector that represents a square of values
+pub struct PixelArray<'a, T> {
+    pixels: &'a [T],
+    width: usize,
+}
+
+impl<'a, T> PixelArray<'a, T> {
+    pub fn new(pixels: &'a [T], width: usize) -> Self {
+        Self { pixels, width }
+    }
+
+    pub fn width(&self) -> usize {
+        self.width
+    }
+}
+
+impl<'a, T> Index<usize> for PixelArray<'a, T> {
+    type Output = T;
+    fn index(&self, index: usize) -> &Self::Output {
+        &self.pixels[index]
+    }
+}
+
+impl<'a, T> Index<Point> for PixelArray<'a, T> {
```

Comment:
Let me just say I love Rust! :heart_eyes:

Being able to index a one-dimensional `Vec` with `(x, y)` is so great. :slightly_smiling_face: